File: app/src/main/java/com/aprivate/sean/ohcg/GameState.java
package com.aprivate.sean.ohcg;

import java.io.Serializable;

public class GameState implements Serializable {

    public GameState(){
        setHandState(HandState.BeginningOfHand);
    }
    private boolean gameInProgress;
    private int handCount;

    public HandState getHandState() {
        return handState;
    }

    public void setHandState(HandState handState) {
        this.handState = handState;
    }

    private HandState handState;

    public boolean isGameInProgress() {
        return gameInProgress;
    }

    public void setGameInProgress(boolean gameInProgress) {
        this.gameInProgress = gameInProgress;
    }


    public int getHandCount() {
        return handCount;
    }

    public void setHandCount(int handCount) {
        this.handCount = handCount;
    }

    public int getDealCount() {
        return dealCount;
    }

    public void setDealCount(int dealCount) {
        this.dealCount = dealCount;
    }

    public int getCurrentDealer() {
        return currentDealer;
    }

    public void setCurrentDealer(int currentDealer) {
        this.currentDealer = currentDealer;
    }

    private int dealCount;
    private int currentDealer;

    public CurrentHandState getCurrentHandState() {
        return currentHandState;
    }

    public void setCurrentHandState(CurrentHandState currentHandState) {
        this.currentHandState = currentHandState;
    }

    private CurrentHandState currentHandState;
}

File: app/src/main/java/com/aprivate/sean/ohcg/GameState.kt
package com.aprivate.sean.ohcg

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey

import java.io.Serializable
import java.sql.Date
import java.sql.Time
import java.time.LocalDate
import java.util.*

class GameState : Serializable {
    @PrimaryKey
    var uid: UUID

    var isGameInProgress: Boolean = false

    var handCount: Int = 0

    var handState: HandState? = null

    var dealCount: Int = 0

    var currentDealer: ScoreBoardItem? = null

    var currentHandState: CurrentHandState? = null

    var players: List<ScoreBoardItem> = emptyList()

    var dateStarted: LocalDate = LocalDate.now()

    init {
        handState = HandState.BeginningOfHand
        uid = UUID.randomUUID()
    }
}
